### PR TITLE
Fix installation of stack on CI and retry nix-setup

### PR DESCRIPTION
**Pull request by junjihashimoto**
This PR tries to fix the setup-error of github-actions(https://github.com/hasktorch/hasktorch/commit/99ec5bd9606dfc7613b352482f74bc857f9aed59/checks ).

### CODE REVIEW

File: .github/workflows/nix-linux.yml [hunk 1]
```diff
@@ -14,10 +14,10 @@ jobs:
         curl https://nixos.org/nix/install | sh
     - name: Setup cachix
       run: |
-        . ~/.nix-profile/etc/profile.d/nix.sh
+        . ~/.nix-profile/etc/profile.d/nix.sh || . ~/.nix-profile/etc/profile.d/nix.sh || . ~/.nix-profile/etc/profile.d/nix.sh || true
```
Comment (tscholak) [inline .github/workflows/nix-linux.yml]: this is a funny change, what does it do?
are you trying to run it three times, giving up in the end?

**Comment by austinvhuang** [comment]
Thanks for the fix @junjihashimoto ! I need to get caught up on github-actions functionality.

File: .github/workflows/nix-linux.yml (hunk 1 above)
Comment (junjihashimoto) [reply .github/workflows/nix-linux.yml]: Yes. It is correct with your understanding.

**Comment by junjihashimoto** [comment]
The error causes that github-actions provides pre-installed ghc, cabal and stack.
The pre-installed ghc's versions include ghc-8.8.
ghc-8.8 provides base-4.13 which conflicts cabal.project.freeze using LTS.

**Pull request merged**